src/factories/meta/CollectionCollector_factory.h

```diff
@@ -6,17 +6,20 @@
 
 namespace eicrecon {
 
-template <class T>
-class CollectionCollector_factory : public JOmniFactory<CollectionCollector_factory<T>> {
+template <class T, bool IsOptional = true>
```

Comment:
```suggestion
template <class T, bool IsOptional = false>
```
Opt-out by default? Even if we want optional in most places, it's good to advertise that this is a flag that can be toggled.